### PR TITLE
Fuse same MaxPools, Pads, etc

### PR DESCRIPTION
**Pull request by mvafin**
### Details:
 - *Fuse same MaxPools, Pads, etc*

### Tickets:
 - *Contributes to 124970*

### CODE REVIEW

File: src/common/transformations/src/transformations/common_optimizations/shared_ops_optimization.cpp [hunk 1]
```diff
@@ -163,6 +170,44 @@ bool converts_are_equal(const Node* lhs, const Node* rhs) {
...
+bool max_pools_8_are_equal(const Node* lhs, const Node* rhs) {
+    const auto l_pool = as_type<const v8::MaxPool>(lhs);
+    const auto r_pool = as_type<const v8::MaxPool>(rhs);
+    if (!l_pool || !r_pool)
+        return false;
+    return l_pool->get_autob() == r_pool->get_autob() && l_pool->get_axis() == r_pool->get_axis() &&
+           l_pool->get_auto_pad() == r_pool->get_auto_pad() && l_pool->get_strides() == r_pool->get_strides() &&
+           l_pool->get_dilations() == r_pool->get_dilations() && l_pool->get_pads_begin() == r_pool->get_pads_begin() &&
+           l_pool->get_pads_end() == r_pool->get_pads_end() && l_pool->get_kernel() == r_pool->get_kernel() &&
+           l_pool->get_rounding_type() == r_pool->get_rounding_type() && l_pool->get_index_element_type() == r_pool->get_index_element_type() &&
```
Comment (mvafin) [inline src/common/transformations/src/transformations/common_optimizations/shared_ops_optimization.cpp:207]:
```suggestion
           l_pool->get_rounding_type() == r_pool->get_rounding_type() &&
           l_pool->get_index_element_type() == r_pool->get_index_element_type() &&
```

File: src/common/transformations/src/transformations/common_optimizations/shared_ops_optimization.cpp [hunk 2]
```diff
@@ -163,6 +170,45 @@ bool converts_are_equal(const Node* lhs, const Node* rhs) {
...
+        return false;
+    return l_pad->get_pad_mode() == r_pad->get_pad_mode() && inputs_from_same_source_or_equal_constants(lhs, rhs);
+}
+
+bool max_pools_8_are_equal(const Node* lhs, const Node* rhs) {
+    const auto l_pool = as_type<const v8::MaxPool>(lhs);
+    const auto r_pool = as_type<const v8::MaxPool>(rhs);
+    if (!l_pool || !r_pool)
+        return false;
+    return l_pool->get_autob() == r_pool->get_autob() && l_pool->get_axis() == r_pool->get_axis() &&
```
Comment (jane-intel) [inline src/common/transformations/src/transformations/common_optimizations/shared_ops_optimization.cpp:207]: I don't see auto broadcasting for Max Pool
```suggestion
    return l_pool->get_axis() == r_pool->get_axis() &&
```

**Review by jane-intel — APPROVED** [review]

File: src/common/transformations/src/transformations/common_optimizations/shared_ops_optimization.cpp [hunk 3]
```diff
@@ -208,6 +260,15 @@ bool pass::SharedOpOptimization::run_on_model(const shared_ptr<Model>& model) {
         RECORD(v1::Reshape, reshapes_are_equal),
         RECORD(v0::ShapeOf, shapeof_are_equal),
         RECORD(v3::ShapeOf, shapeof_are_equal),
+        RECORD(v1::Divide, divide_are_equal),
+        RECORD(v1::Multiply, eltwise_arithmethic_are_equal),
```
Comment (itikhono) [inline src/common/transformations/src/transformations/common_optimizations/shared_ops_optimization.cpp:264]: I would recommend to run a performance validation

Comment (jane-intel) [reply src/common/transformations/src/transformations/common_optimizations/shared_ops_optimization.cpp:264]: If there will be an activity to measure performance -- please include these changes with Transpose also:
https://github.com/openvinotoolkit/openvino/pull/18775/

**Review by itikhono — APPROVED** [review]

**Comment by itikhono** [comment]
DL Bench (Benchmark app) validation showed no significant regressions on DLB ModelZoo scope.
Approved.

File: src/common/transformations/src/transformations/common_optimizations/shared_ops_optimization.cpp [hunk 4]
```diff
@@ -164,6 +175,44 @@ bool converts_are_equal(const Node* lhs, const Node* rhs) {
...
+    if (!l_pool || !r_pool)
+        return false;
+    return l_pool->get_axis() == r_pool->get_axis() && l_pool->get_auto_pad() == r_pool->get_auto_pad() &&
+           l_pool->get_strides() == r_pool->get_strides() && l_pool->get_dilations() == r_pool->get_dilations() &&
+           l_pool->get_pads_begin() == r_pool->get_pads_begin() && l_pool->get_pads_end() == r_pool->get_pads_end() &&
+           l_pool->get_kernel() == r_pool->get_kernel() && l_pool->get_rounding_type() == r_pool->get_rounding_type() &&
+           l_pool->get_index_element_type() == r_pool->get_index_element_type() &&
+           inputs_from_same_source_or_equal_constants(lhs, rhs);
+}
+
```
Comment (mvafin) [inline src/common/transformations/src/transformations/common_optimizations/shared_ops_optimization.cpp:215]:
```suggestion

```

**Pull request merged**